shared/src/main/scala/pl/metastack/metatime/Offset.scala

```diff
@@ -15,6 +17,34 @@ object Formatter {
   def branch(ranges: Seq[(Range, Any)], fallback: Any): Formatter = ???
 }
 
-trait Offset[T] {
-  def format: String = ???
+trait Offset[+T] {
```

Comment:
The problem is that you define a generic type parameter `Component` in line 47 which is not a sub-type of the trait `Component`. The generic has the same name, but doesn't actually point to the `Component` trait as you may expect.

This works:

```scala
def apply(value: Component): Offset[Component] = new Offset[Component] { ... }
```

Or even better:

```scala
def apply[T <: Component](value: T): Offset[T] = new Offset[T] { ... }
```